slam_toolbox/src/slam_toolbox_common.cpp

```diff
@@ -446,8 +447,9 @@ bool SlamToolbox::shouldProcessScan(
 
   // check moved enough, within 10% for correction error
   const double sq_dist_to_last_accepted_pose = last_pose.SquaredDistance(pose);
+  const double head_to_last_accepted_pose = fmod(fabs(pose.GetHeading() - last_pose.GetHeading()), M_PI) / M_PI * 180;
```

Comment:
This is incorrect, if you have angles on different sides of the unit circle then simply comparing distance will not be representative of the actual angle. fmoding that is not the same thing as the actual angular distance.

Reply:
I have checked and modified.